src/parser/typescript/analyze/index.ts

```diff
@@ -215,6 +215,76 @@ function analyzeDollarDollarVariables(
     );
   }
 
+  addSvelte5Globals();
+
+  function addSvelte5Globals() {
+    for (const svelte5Global of globalsForSvelte5) {
+      switch (svelte5Global) {
+        case "$state": {
+          if (
+            scopeManager.globalScope!.through.some(
+              (reference) => reference.identifier.name === svelte5Global,
+            )
+          ) {
```

Comment:
By moving this judgment before the `switch`, I think we can eliminate the four same implementations.

Reply:
Ohhhhh...
https://github.com/sveltejs/svelte-eslint-parser/pull/425/commits/0a61e915bfe1ebb9f63cc658e8fa0720cb482b3c